src/test/java/org/springframework/samples/petclinic/owner/PetTests.java

```diff
@@ -0,0 +1,81 @@
+package org.springframework.samples.petclinic.owner;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.samples.petclinic.visit.Visit;
+
+import java.util.List;
+import java.util.Set;
+import java.util.LinkedHashSet;
+import java.time.*;
+import java.util.Date;
+
+public class PetTests {
+
+	private Pet pet;
+	private Date birthDate;
+
+	private void establishCurrentDateForTesting() {
+		LocalDateTime timePoint = LocalDateTime.now();
+		LocalDate localDate = timePoint.toLocalDate();
+		birthDate = Date.from(localDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
+	}
+
+	@Before
+	public void testSetUp() {
+		this.pet = new Pet();
+		PetType dog = new PetType();
+		dog.setId(9);
+		dog.setName("Duncan Jones");
+	}
+
+	@Test
+	public void testSetAndGetBirthDate() {
+		//Converting the current time to a local date and ultimately a date to be input into setBirthDate
+		this.establishCurrentDateForTesting();
+		pet.setBirthDate(this.birthDate);
+		Date resultOfGetDate = pet.getBirthDate();
+		assertEquals(this.birthDate, resultOfGetDate);
+	}
+
+	@Test
+	public void testSetAndGetType() {
+		//Creating a new pet type to test the setters and getters for pet's type
+		PetType walrus = new PetType();
+		walrus.setId(36);
+		walrus.setName("Alex Garland");
+		pet.setType(walrus);
+		PetType resultOfGetType = pet.getType();
+		assertEquals(walrus.getName(), resultOfGetType.getName());
+	}
+
+	@Test
+	public void testSetAndGetOwner() {
+		//Creating a new owner type to test the setters and getters for the pet's owner
+		Owner amandeepBhandal = new Owner();
+		pet.setOwner(amandeepBhandal);
+		Owner resultOfGetOwner = pet.getOwner();
+		assertEquals(amandeepBhandal.getAddress(), resultOfGetOwner.getAddress());
```

Comment:
Is the decision to use `getAddress` arbitrary? Also, if using `getAddress` to check if the owners are the same, consider setting an actual explicit address value (like "123 Fake Address") and asserting that it's equal.

Reply:
Will refactor to account for all the attributes of the owner as well as actually setting the values. Good catch.